app/models/concerns/alegre_v2.rb

```diff
@@ -148,28 +159,32 @@ def get_type(project_media)
       type
     end
 
+    def content_hash_for_value(value)
+      value.nil? ? nil : Digest::MD5.hexdigest(value)
+    end
+
     def content_hash(project_media, field)
       if Bot::Alegre::ALL_TEXT_SIMILARITY_FIELDS.include?(field)
-        Digest::MD5.hexdigest(project_media.send(field))
+        content_hash_for_value(project_media.send(field))
+      elsif project_media.is_link?
+        return content_hash_for_value(project_media.media.url)
+      elsif project_media.is_a?(TemporaryProjectMedia)
+        return Rails.cache.read("url_sha:#{project_media.url}")
+      elsif !project_media.is_text?
+        return project_media.media.file.filename.split(".").first
```

Comment:
Suggestion: It's stronger to change this validation to something like `is_uploaded_media?`, which basically checks if it's audio, image or video. I'm saying this because we also have `blank` and may have other types.